dvc/dvcfile.py

```diff
@@ -144,8 +145,8 @@ def _load(self):
         # 3. path doesn't represent a regular file
         # 4. when the file is git ignored
         if not self.exists():
-            is_ignored = self.repo.fs.exists(self.path, use_dvcignore=False)
-            raise StageFileDoesNotExistError(self.path, dvc_ignored=is_ignored)
+            file_exist = self.repo.fs.exists(self.path)
+            raise StageFileDoesNotExistError(self.path, dvc_ignored=file_exist)
```

Comment:
This should not look like that. In `exists` we have two statements (`exists and is_not_dvcignored`). Here we check `exists` and infer the information whether file is ignored. But to infer this knowledge we need to know how `exists` works. We could simply check here again if path is `dvcignored` and provide that into error constructor.